Microsoft.HealthVault.Fhir/Transformers/AllergyIntoleranceToHealthVault.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Get Real Health.  All rights reserved.
+// MIT License
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Linq;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Support;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.ItemTypes;
+using static Hl7.Fhir.Model.AllergyIntolerance;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static class AllergyIntoleranceToHealthVault
+    {
+        public static Allergy ToHealthVault(this AllergyIntolerance allergyIntolerance)
+        {
+            if (allergyIntolerance.Code.IsNullOrEmpty())
+            {
+                throw new System.ArgumentException($"Can not transform a {typeof(AllergyIntolerance)} with no code into {typeof(Allergy)}");
+            }
+
+            var allergy = allergyIntolerance.ToThingBase<Allergy>();
+
+            string allergenType = string.Empty;
+            var allergyExtension = allergyIntolerance.GetExtension(HealthVaultExtensions.Allergy);
+            if (allergyExtension != null)
+            {
+                allergenType = allergyExtension.GetStringExtension(HealthVaultExtensions.AllergenType);
+                allergy.AllergenCode = allergyExtension.GetExtensionValue<CodeableConcept>(HealthVaultExtensions.AllergenCode)?.ToCodableValue();
+                allergy.Treatment = allergyExtension.GetExtensionValue<CodeableConcept>(HealthVaultExtensions.AllergyTreatement)?.ToCodableValue();
+
+            }
+
+            foreach (var code in allergyIntolerance.Code.Coding)
```

Comment:
Once we set this once, do we want to continue to looping through additional codings and possibly overwriting it?